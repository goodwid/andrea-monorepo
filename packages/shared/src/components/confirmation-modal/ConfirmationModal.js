import React from 'react';
import PropTypes from 'prop-types';
import styles from './ConfirmationModal.scss';

import { MdClose as Close } from 'react-icons/md';
// import { Button } from '@material-ui/core';

const ConfirmationModal = props => {
  const { handleConfirmationClose, handleNoClicked, handleYesClicked } = props;

  return (
    <article className={styles.ConfirmationModal}>
      <div className="overlay" onClick={handleConfirmationClose} />
      <aside className="modal-body">
        <button onClick={handleConfirmationClose} className="close-button">
          <Close />
        </button>
        <main className="content">
          <h2 className="question">Are you sure?</h2>
          <section className="selection-buttons">
            <button className="btn-approve" onClick={handleYesClicked}>
              YES
            </button>
            <button className="btn-decline" onClick={handleNoClicked}>
              NO
            </button>
          </section>
        </main>
      </aside>
    </article>
  );
};

ConfirmationModal.propTypes = {
  handleConfirmationClose: PropTypes.func.isRequired,
  handleYesClicked: PropTypes.func.isRequired,
  handleNoClicked: PropTypes.func.isRequired,
};

export default ConfirmationModal;
