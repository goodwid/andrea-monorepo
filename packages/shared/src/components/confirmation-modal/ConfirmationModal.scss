@import 'shared/styles/variables.scss';

:local(.ConfirmationModal) {
  font-family: $default-font;

  .overlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    background-color: $default-modal-overlay-color;
    z-index: 280;
    animation: fade-in .1s ease-in forwards;
  }

  .modal-body {
    position: fixed;
    top: 50%;
    left: 50%;
    padding: 0;
    width: 80vw;
    max-width: 30rem;
    height: 20rem;
    min-height: 5rem;
    background-color: $default-modal-background;
    font-size: 1.6rem;
    z-index: 290;
    animation: modal-in .3s ease forwards;
    transform: translate(-50%, -50%);
    border-radius: 0.5rem;
    box-shadow: 0.2rem 0.2rem 2rem 0.25rem $default-modal-shadow-color;

    .close-button {
      position: absolute;
      top: 0.3rem;
      left: 0.3rem;
      font-size: 2rem;
      font-weight: 300;
      z-index: 4060;
      height: auto;
      padding: 0;
    }

    .content {
      display: flex;
      justify-content: flex-start;
      flex-direction: column;
      position: absolute;
      top: 3rem;
      left: 0;
      width: 100%;
      text-align: center;

      .question {
        font-size: 3.8rem;
        font-weight: 500;
        line-height: 4rem;
      }

      .selection-buttons {
        position: fixed;
        bottom: 1.75rem;
        width: 100%;
        height: 5rem;
        background-color: $default-modal-background;
        display: flex;
        justify-content: space-between;
        align-items: center;

        button {
          padding: 1.5rem;
          margin: 1rem;
          border-width: 4px;
          border-style: solid;
          width: 45%;
          max-width: 200px;
          font-size: 1.8rem;
        }

        .btn-approve {
          @include selection-btn-color($confirmation-button-color);
        }

        .btn-decline {
          @include selection-btn-color($warning-button-color);
        }
      }
    }
  }
}
