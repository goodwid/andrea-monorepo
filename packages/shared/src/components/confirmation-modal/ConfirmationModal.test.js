import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationModal from './ConfirmationModal';

describe('ConfirmationModal page', () => {
  it('matches the snapshot', () => {
    const handleConfirmationClose = jest.fn();
    const handleYesClicked = jest.fn();
    const handleNoClicked = jest.fn();
    const wrapper = shallow(
      <ConfirmationModal handleConfirmationClose={handleConfirmationClose}
        handleNoClicked={handleNoClicked}
        handleYesClicked={handleYesClicked}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
