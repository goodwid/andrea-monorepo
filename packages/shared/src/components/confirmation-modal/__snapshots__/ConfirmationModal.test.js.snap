// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal page matches the snapshot 1`] = `
<article
  className="ConfirmationModal"
>
  <div
    className="overlay"
    onClick={[MockFunction]}
  />
  <aside
    className="modal-body"
  >
    <button
      className="close-button"
      onClick={[MockFunction]}
    >
      <MdClose />
    </button>
    <main
      className="content"
    >
      <h2
        className="question"
      >
        Are you sure?
      </h2>
      <section
        className="selection-buttons"
      >
        <button
          className="btn-approve"
          onClick={[MockFunction]}
        >
          YES
        </button>
        <button
          className="btn-decline"
          onClick={[MockFunction]}
        >
          NO
        </button>
      </section>
    </main>
  </aside>
</article>
`;
